tap_snowflake/client.py

```diff
@@ -76,12 +102,131 @@ def discover_catalog_entries(self) -> list[dict]:
 
         return result
 
+    def get_table_profile(
+        self,
+        full_table_name: str,
+        stats: set[ProfileStats],
+        profile_columns: list[str] | None = None,
+    ) -> TableProfile:
```

Comment:
Most immediately, this new `get_table_profile()` implementation would give max values for the replication key column. However, it can also be used in the future to put rowcount estimates into the catalog.

In the future, the count of 'null column values' metadata can be used to validate if a column is valid as a primary key and/or replication key - since nulls in either of these columns would be a warning if not an error.

Since we definitely don't want to rely on pulling stats with multiple passes if not needed, I felt it was best to go ahead and build a single mechanism we could use to optionally query for other stats at the same time if there's need to do so.